flexmeasures/data/services/scheduling.py

```diff
@@ -242,9 +216,25 @@ def load_custom_scheduler(scheduler_specs: dict) -> Tuple[Scheduler, dict]:
     # get scheduling function
     assert hasattr(
         module, scheduler_specs["class"]
-    ), "Module at {module_descr} has no class {scheduler_specs['class']}"
+    ), f"Module at {module_descr} has no class {scheduler_specs['class']}"
 
     scheduler_class = getattr(module, scheduler_specs["class"])
+    schedule_function_name = "compute_schedule"
+    if not hasattr(scheduler_class, schedule_function_name):
+        raise NotImplementedError(
+            f"No function {schedule_function_name} in {scheduler_class}. Cannot load custom scheduler."
+        )
+    return scheduler_class
+
+
+def get_data_source_info_by_scheduler_class(scheduler_class: type) -> dict:
```

Comment:
I agree that this logic should live in the base `Scheduler` class.

I don't want to have to initialize a scheduler only to get to the data source, though. `get_data_source_for_job` is currently the one we need, and it's used when we GET a schedule.

Reply:
I think you don't need to initialize an instance to read a class attribute. So `Scheduler.data_source_info` rather than `Scheduler().data_source_info`.